packages/@aws-cdk/aws-apigateway/test/test.domains.ts

```diff
@@ -267,6 +267,20 @@ export = {
     test.done();
   },
 
+  'domain name cannot contain uppercase letters'(test: Test) {
+    // GIVEN
+    const stack = new Stack();
+    const certificate = new acm.Certificate(stack, 'cert', { domainName: 'someDomainWithUpercase.domain.com' });
+
+    // WHEN
+    test.throws(() => {
+      new apigw.DomainName(stack, 'someDomain', {domainName: 'someDomainWithUpercase.domain.com', certificate});
+    });
```

Comment:
check for the expected exception and not something different

```suggestion
    test.throws(() => {
      new apigw.DomainName(stack, 'someDomain', {domainName: 'someDomainWithUpercase.domain.com', certificate});
    }, /uppercase/);
```